hub/config/src/main/java/uk/gov/ida/hub/config/application/CertificateService.java

```diff
@@ -1,119 +1,107 @@
+//    private static final Logger LOG = LoggerFactory.getLogger(CertificateService.class);
 
     @Inject
     public CertificateService(
-            ConfigRepository<TransactionConfig> transactionDataSource,
-            ConfigRepository<MatchingServiceConfig> matchingServiceDataSource,
+            ConfigRepository<TransactionConfig> transactionConfigRepository,
+            ConfigRepository<MatchingServiceConfig> matchingServiceConfigRepository,
             CertificateValidityChecker certificateValidityChecker) {
-        this.transactionDataSource = transactionDataSource;
-        this.matchingServiceDataSource = matchingServiceDataSource;
+        this.configRepositories.add((ConfigRepository<T>)transactionConfigRepository);
+        this.configRepositories.add((ConfigRepository<T>)matchingServiceConfigRepository);
         this.certificateValidityChecker = certificateValidityChecker;
-
-        getTransactionEncryptionCert = encryptiondataSource.apply(transactionDataSource, FederationEntityType.RP);
-        getMatchingServiceEncryptionCert = encryptiondataSource.apply(matchingServiceDataSource, FederationEntityType.MS);
-        getTransactionSignatureCert = signatureDataSource.apply(transactionDataSource, FederationEntityType.RP);
-        getMatchingServiceSignatureCert = signatureDataSource.apply(matchingServiceDataSource, FederationEntityType.MS);
     }
 
-    public CertificateDetails encryptionCertificateFor(String entityId) {
-        CertificateDetails certificateDetails = getTransactionEncryptionCert.apply(entityId)
-                .orElseGet(() -> getMatchingServiceEncryptionCert.apply(entityId)
-                .orElseThrow(() -> new NoCertificateFoundException()));
+    public Set<CertificateDetails> getAllCertificateDetails() {
+        return configRepositories.stream()
+                .flatMap(this::getAllCertificateDetails)
+                .collect(Collectors.toSet());
+    }
 
-        if (certificateDetails.isNotEnabled()) {
+    public  CertificateDetails encryptionCertificateFor(String entityId) {
+        T config = getConfig(entityId);
+        CertificateDetails certDetails = createCertificateDetails(config, config.getEncryptionCertificate());
+        if (!certificateValidityChecker.isValid(certDetails)){
+            throw new NoCertificateFoundException();
+        }
+        if (certDetails.isNotEnabled()){
             throw new CertificateDisabledException();
         }
+        return certDetails;
+    }
 
-        return certificateDetails;
+    public List<CertificateDetails> signatureVerificationCertificatesFor(String entityId) {
+        T config = getConfig(entityId);
+        return config.getSignatureVerificationCertificates()
+                .stream()
+                .map(cert -> createCertificateDetails(config, cert))
+                .filter(cd -> certificateValidityChecker.isValid(cd))
+                .collect(Collectors.collectingAndThen(Collectors.toList(), this::throwIfEmpty));
```

Comment:
This will throw if the list is ultimately empty - which seems sensible given the other throws. Whereas, if I'm deciphering the existing code correctly, it was happy to return a list containing only nulls. So wanted to flag up that the behaviour might have slightly changed. Maybe?

Reply:
Yes, not sure if the existing code would return a list of nulls, but that's not good behaviour. The existing tests did check that an empty list should throw.